app/models/mission/MissionTable.scala

```diff
@@ -265,10 +265,15 @@ object MissionTable {
     *
     * @param userId User's UUID
     * @param includeOnboarding should any onboarding missions be included
+    * @param includeSkipped should any skipped missions be included
     */
-  def selectCompletedMissionsByAUser(userId: UUID, includeOnboarding: Boolean): List[Mission] = db.withSession { implicit session =>
+  def selectCompletedMissionsByAUser(userId: UUID, includeOnboarding: Boolean, includeSkipped: Boolean): List[Mission] = db.withSession { implicit session =>
     val _missions = if (includeOnboarding) {
-      missions.filter(m => m.userId === userId.toString && m.completed)
+      if (includeSkipped) {
```

Comment:
Hmm. Although it is technically true right now that you can't skip a mission other than a tutorial mission, I don't think that we should make that assumption in this function. So really, the conditional should also be in the `else` case where we are excluding onboarding missions. Since we want to do that check no matter what, I think this function could be written more concisely as:
```
val _m1 = missions.filter(m => m.userId === userId.toString && m.completed)
val _m2 = if (includeOnboarding) _m1 else _m1.filterNot(_.missionTypeId inSet MissionTypeTable.onboardingTypeIds)
val _m3 = if (includeSkipped) _m2 else _m2.filterNot(_.skipped)

_m3.list.groupBy(_.missionId).map(_._2.head).toList
```

Reply:
Yes, sounds good